users/views.py

```diff
@@ -0,0 +1,66 @@
+import json
+from django.views import View
+from .models      import User
+from products.models import Product
+from django.http  import JsonResponse
+import bcrypt
+import jwt
+import django
+import re
+import my_settings
+
+class SignUpView(View):
+    def post(self , request):
+        try:
+            data = json.loads(request.body)
+            token =request.headers.get('User-Agent')
+            first_name = data["first_name"]
+            last_name  = data["last_name"]
+            birth_date = data['birth_date']
+            email      = data["email"]
+            password   = data["password"]
+
+            pattern=r'[A-Z0-9._%+-]+@[A-Z0-9,-]+\.[A-Z]{2,4}'
+            regex=re.compile(pattern,flags=re.IGNORECASE)
+            
+            if len(regex.findall(email)) == 0:
+                return JsonResponse({"message":"EMAIL_INVALID"})
+            
+            if len(password)<8 :
+                return JsonResponse({"message": "PASSWORD_LENGTH_ERROR"})
+
+            hash_password   = bcrypt.hashpw(password.encode('utf-8'),bcrypt.gensalt())
+            hash_password_d = hash_password.decode('utf-8')
```

Comment:
```suggestion
            hash_password   = bcrypt.hashpw(password.encode('utf-8'),bcrypt.gensalt()).decode('utf-8')
```
두 번에 분리하지 말고, 한 번에 진행하셔도 됩니다 :)